content/docs/command-reference/remote/modify.md

```diff
@@ -333,14 +333,21 @@ following parameters (listed in order or precedence) to customize the
 authentication method:
 
 1. `connection_string` is used for authentication if given (all others params
-   are ignored).
+   are ignored). If `connection_string` is not set, the `account_name` must be
+   given.
 2. If `tenant_id` and `client_id` or `client_secret` are given, Active Directory
```

Comment:
> If `connection_string` is not set, the `account_name` must be given.

@isidentical do `tenant_id`, `client_id`, `client_secret` require `account_name`? That's kind of what it looks like from the change above. Can you explain so we can take this over? Thanks

Reply:
Yes, they do. Basically, you should either have `connection_string` or `account_name` set all the time; https://github.com/dask/adlfs/blob/96958331854c5a67917d4c9a802a62720d091374/adlfs/spec.py#L493-L526

Reply:
OK thanks @isidentical but it would still be great if you can elaborate because the doc says "DVC authenticates using an Azure default credential" first, and then the list says that either `connection_string` or `account_name` must be given. Which one is it? Thanks

Reply:
> Which one is it? Thanks

In either case (even with default credentials) the `account_name` should be given. Previously whether the user wanted anonymous login or not was a bit ambiguous since both default credentials and anonymous login wanted it so we tried to infer it in various terms but now it is done by an explicit option so if you only specify `account_name` then you will use default credentials but if you also specify `allow_anonymous_login` then you will use anonymous login.